sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasingSuite.scala

```diff
@@ -493,6 +491,156 @@ class NestedColumnAliasingSuite extends SchemaPruningTest {
     comparePlans(optimized3, expected3)
   }
 
+  test("Nested field pruning for Window") {
+    val spec = windowSpec($"address" :: Nil, $"id".asc :: Nil, UnspecifiedFrame)
+    val winExpr = windowExpr(RowNumber(), spec)
+
+    val query1 = contact
+      .select($"name.first", winExpr.as('window))
+      .analyze
+    val optimized1 = Optimize.execute(query1)
+    val expected1 = contact
+      .select($"name.first", $"address", $"id")
+      .window(Seq(winExpr.as("window")), Seq($"address"), Seq($"id".asc))
+      .select($"first", $"window")
+      .analyze
+    comparePlans(optimized1, expected1)
+
+    val query2 = contact
+      .select($"name.first", winExpr.as('window))
+      .orderBy($"name.last".asc)
+      .analyze
+    val optimized2 = Optimize.execute(query2)
+    val aliases2 = collectGeneratedAliases(optimized2)
+    val expected2 = contact
+      .select($"name.first", $"address", $"id", $"name.last".as(aliases2(1)))
+      .window(Seq(winExpr.as("window")), Seq($"address"), Seq($"id".asc))
+      .select($"first", $"window", $"${aliases2(1)}".as(aliases2(0)))
+      .orderBy($"${aliases2(0)}".asc)
+      .select($"first", $"window")
+      .analyze
+    comparePlans(optimized2, expected2)
+  }
+
+  test("Nested field pruning for Filter") {
```

Comment:
It is not for only Filter. Maybe `Nested field pruning for Filter with other operators`?

Reply:
Sure.

Reply:
Name changed.